Index: android-cropimage/src/com/android/camera/CropImage.java
===================================================================
--- android-cropimage.orig/src/com/android/camera/CropImage.java	2011-07-20 21:47:57.882158360 +0200
+++ android-cropimage/src/com/android/camera/CropImage.java	2011-07-20 21:52:12.882168456 +0200
@@ -97,8 +97,6 @@
 
         mImageView = (CropImageView) findViewById(R.id.image);
 
-        MenuHelper.showStorageToast(this);
-
         Intent intent = getIntent();
         Bundle extras = intent.getExtras();
 
@@ -131,20 +129,6 @@
         }
 
         if (mBitmap == null) {
-            Uri target = intent.getData();
-            mAllImages = ImageManager.makeImageList(mContentResolver, target,
-                    ImageManager.SORT_ASCENDING);
-            mImage = mAllImages.getImageForUri(target);
-            if (mImage != null) {
-                // Don't read in really large bitmaps. Use the (big) thumbnail
-                // instead.
-                // TODO when saving the resulting bitmap use the
-                // decode/crop/encode api so we don't lose any resolution.
-                mBitmap = mImage.thumbBitmap(IImage.ROTATE_AS_NEEDED);
-            }
-        }
-
-        if (mBitmap == null) {
             finish();
             return;
         }
@@ -344,49 +328,6 @@
                 Log.e(TAG, "Failed to set wallpaper.", e);
                 setResult(RESULT_CANCELED);
             }
-        } else {
-            Bundle extras = new Bundle();
-            extras.putString("rect", mCrop.getCropRect().toString());
-
-            File oldPath = new File(mImage.getDataPath());
-            File directory = new File(oldPath.getParent());
-
-            int x = 0;
-            String fileName = oldPath.getName();
-            fileName = fileName.substring(0, fileName.lastIndexOf("."));
-
-            // Try file-1.jpg, file-2.jpg, ... until we find a filename which
-            // does not exist yet.
-            while (true) {
-                x += 1;
-                String candidate = directory.toString()
-                        + "/" + fileName + "-" + x + ".jpg";
-                boolean exists = (new File(candidate)).exists();
-                if (!exists) {
-                    break;
-                }
-            }
-
-            try {
-                int[] degree = new int[1];
-                Uri newUri = ImageManager.addImage(
-                        mContentResolver,
-                        mImage.getTitle(),
-                        mImage.getDateTaken(),
-                        null,    // TODO this null is going to cause us to lose
-                                 // the location (gps).
-                        directory.toString(), fileName + "-" + x + ".jpg",
-                        croppedImage, null,
-                        degree);
-
-                setResult(RESULT_OK, new Intent()
-                        .setAction(newUri.toString())
-                        .putExtras(extras));
-            } catch (Exception ex) {
-                // basically ignore this or put up
-                // some ui saying we failed
-                Log.e(TAG, "store image fail, continue anyway", ex);
-            }
         }
 
         final Bitmap b = croppedImage;
